File: Example/ObjcExample/ObjcExample/CollectionViewDemoController.m
//
//  CollectionViewDemoController.m
//  ObjcExample
//
//  Created by Animax Deng on 5/29/20.
//  Copyright © 2020 Animax. All rights reserved.
//

#import "CollectionViewDemoController.h"
#import <A_StyleInjection/A_StyleInjection.h>

@interface DemoCollectionCell : UICollectionViewCell

@property (weak, nonatomic) IBOutlet UILabel *label;
@property (weak, nonatomic) IBOutlet UITextField *input;

@end

@implementation DemoCollectionCell

@end


@interface CollectionViewDemoController () <UICollectionViewDataSource, UICollectionViewDelegate>

@property (weak, nonatomic) IBOutlet UICollectionView *collectionView;

@end

@implementation CollectionViewDemoController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.collectionView.dataSource = self;
    self.collectionView.delegate = self;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return 500;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    DemoCollectionCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"cell" forIndexPath:indexPath];
    [cell.label setText:[NSString stringWithFormat:@"%ld", (long)indexPath.row]];
    return cell;
}

- (IBAction)onClickSetStyle:(id)sender {
    [[A_StyleManager shared] setupStyleSourceRepository:[[StylePlistProvider alloc] init:@"StyleSheet"]];
}

@end

File: Example/SwiftExample/SwiftExample/CollectionViewDemoController.swift
//
//  CollectionViewDemoController.swift
//  SwiftExample
//
//  Created by Animax Deng on 5/31/20.
//  Copyright © 2020 Animax. All rights reserved.
//

import UIKit
import A_StyleInjection

class DemoCollectionCell: UICollectionViewCell {
    @IBOutlet weak var label: UILabel!
    @IBOutlet weak var input: UITextField!
}

class CollectionViewDemoController: UIViewController, UICollectionViewDataSource, UICollectionViewDelegate {

    @IBOutlet weak var collectionView: UICollectionView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.collectionView.delegate = self
        self.collectionView.dataSource = self
    }
    
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        
        return 500
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! DemoCollectionCell
        
        cell.label.text = "\(indexPath.row)"
        
        return cell
    }

    @IBAction func onClickSetStyle(_ sender: Any) {
        A_StyleManager.shared().setupStyleSourceRepository(StylePlistProvider("StyleSheet"))
    }
}
